generated/nidaqmx/task.py

```diff
@@ -685,8 +726,32 @@ def register_every_n_samples_acquired_into_buffer_event(
                 Passing None for this parameter unregisters the event callback
                 function.
         """
-        self._interpreter.register_every_n_samples_event(self._handle, EveryNSamplesEventType.ACQUIRED_INTO_BUFFER.value,
-                sample_interval, 0, callback_method, None)
+        if callback_method is not None:
+            if use_new_signature:
+                # New signature: callback(task, every_n_samples_event_type, number_of_samples) -> None
+                def call_every_n_samples_event_callback(task_handle, every_n_samples_event_type, number_of_samples, callback_data):
+                    callback_method(self, EveryNSamplesEventType(every_n_samples_event_type), number_of_samples)
+                    return 0
```

Comment:
Idea: use [`inspect.signature`](https://docs.python.org/3/library/inspect.html#inspect.signature) to detect whether the callback has the old or new signature, based on the number of `POSITIONAL_OR_KEYWORD` arguments.